packages/navigator/src/browser/navigator-decorator-service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, named } from 'inversify';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { ContributionProvider } from '@theia/core/lib/common/contribution-provider';
+import { TreeDecorator, DecorationData, TreeDecoratorService } from '@theia/core/lib/browser/tree/tree-decorator';
+
+/**
+ * Decorator service which emits events from all known tree decorators and caches the current state.
+ */
+@injectable()
+export class NavigatorDecoratorService implements TreeDecoratorService {
+
+    protected emitter: Emitter<Map<string, DecorationData[]>>;
+    protected decorations: Map<string, Map<string, DecorationData>>;
+
+    constructor(@inject(ContributionProvider) @named(TreeDecorator) protected readonly decorators: ContributionProvider<TreeDecorator>) {
```

Comment:
Is it so that there could be multiple `TreeDecoratorService` impls, but only one set of `TreeDecorator`s that would be used among them?